### ISSUE TITLE
[spark] esRDD doesn't respect config setting "es.field.read.empty.as.null"

### ISSUE THREAD

**Issue by MLnick** [elastic/elasticsearch-hadoop#402]
A simple reproduction for numeric fields:

```
object TestES extends App {
  import org.elasticsearch.spark._
  val conf = new SparkConf().setAppName("test").setMaster("local[2]")
  conf.set("es.field.read.empty.as.null", "true")

  val sc = new SparkContext(conf)
  val data = Seq(
    Map("field1" -> 5.4, "field2" -> "foo"), 
    Map("field2" -> "bar"), 
    Map("field1" -> "", "field2" -> "baz")
  )
  sc.makeRDD(data).saveToEs("test/test")

  val rdd = EsSpark.esRDD(sc, "test/test")
  println(rdd.collect.mkString(","))
}
```

Throws:
```
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 0.0 failed 1 times, most recent failure: Lost task 2.0 in stage 0.0 (TID 2, localhost): java.lang.NumberFormatException: empty String
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1020)
	at java.lang.Double.parseDouble(Double.java:540)
	at scala.collection.immutable.StringLike$class.toDouble(StringLike.scala:232)
	at scala.collection.immutable.StringOps.toDouble(StringOps.scala:31)
	at org.elasticsearch.spark.serialization.ScalaValueReader.parseDouble(ScalaValueReader.scala:82)
	at org.elasticsearch.spark.serialization.ScalaValueReader$$anonfun$doubleValue$1.apply(ScalaValueReader.scala:81)
	at org.elasticsearch.spark.serialization.ScalaValueReader$$anonfun$doubleValue$1.apply(ScalaValueReader.scala:81)
	at org.elasticsearch.spark.serialization.ScalaValueReader.checkNull(ScalaValueReader.scala:55)
	at org.elasticsearch.spark.serialization.ScalaValueReader.doubleValue(ScalaValueReader.scala:81)
	at org.elasticsearch.spark.serialization.ScalaValueReader.readValue(ScalaValueReader.scala:40)
	at org.elasticsearch.hadoop.serialization.ScrollReader.parseValue(ScrollReader.java:221)
```

**Comment by MLnick** [comment]
@costin can you take a look at this? It's really causing me a lot of pain and forcing me at the moment to use the InputFormat directly and write my own conversions from Writables -> Scala types :(

It also does not work when I set that config option in `sc.hadoopConfiguration` or in a `Map` passed into `esRDD`

**Comment by costin** [comment]
Hi,
Fixed the issue in master. It was an ... embarrassing bug where the Scala syntax again played tricks on me (the empty/null value was properly parsed but instead of being returned, the conversion continued).

I have uploaded into maven the new dev artifacts.

**Comment by costin** [comment]
Last but not least, thanks for the detailed report and in particular your patience.

**Comment by MLnick** [comment]
Great thanks - will check and confirm against the snapshot build.




If I had more time I would have tried to put together a PR :) thanks again



—

**Comment by swatisaini** [comment]
Hi Costin, 
Is this issue fixed in spark 2.0.2 as well ?
